ast/unique/Unique.go

```diff
@@ -15,30 +15,11 @@
 package unique
 
 import (
-	"fmt"
-	"sync"
-	"time"
-)
-
-var (
-	offset int64 = 0
-	lastMS int64 = 0
-	mutex  sync.Mutex
+	"github.com/rs/xid"
 )
 
 // NewID will create a new unique ID string for this runtime.
 // Uniqueness between system or apps is not necessary.
 func NewID() string {
-	mutex.Lock()
-	defer mutex.Unlock()
-	millisUnix := time.Now().Unix()
-	if lastMS == millisUnix {
-		offset++
-
-		return fmt.Sprintf("%d-%d", lastMS, offset)
-	}
-	lastMS = millisUnix
-	offset = 0
-
-	return fmt.Sprintf("%d-%d", lastMS, offset)
+	return xid.New().String()
```

Comment:
What is the background for this new `NewID()` implementation?

Reply:
@rifqifatih I did this change when testing the _spawn evaluation goroutine on-demand_ changes back then, but keeping it for the improvement since I'm thinking grule's `NewID` can be improved, the reason is:

- it's using mutex. It may be not have direct effect for now in Kuber but I'm thinking that it's unnecessary, can rely on other fast unique string library instead (like xid that I use)
- it's making extra allocation during rule loading. Imagine that there are a ton of component in the rule and it'll have spammy allocs

did quick benchmark - current `NewID` is slower than the new changes (and more alloc also).

Before:

<img width="708" alt="image" src="https://github.com/rezdatiket/grule-rule-engine/assets/107461205/51fba745-f4c2-4b8a-9abd-fc718c856f48">

After:

<img width="715" alt="image" src="https://github.com/rezdatiket/grule-rule-engine/assets/107461205/74ece614-d03d-4c0c-a743-fff5c84f7113">

Reply:
https://github.com/rs/xid#benchmark